centosplus patch

Ref: ELRepo patch  elrepo-fusion-mptsas-mptspi-el8.patch

Revert the changes RH made in their el8 kernel

diff -Naurp rhel8-4.18.0-80/drivers/message/fusion/mptsas.c kernel-4.20.18/drivers/message/fusion/mptsas.c
--- a/drivers/message/fusion/mptsas.c	2019-03-13 11:04:53.000000000 +0000
+++ b/drivers/message/fusion/mptsas.c	2018-11-21 08:22:14.000000000 +0000
@@ -5290,9 +5290,6 @@ mptsas_probe(struct pci_dev *pdev, const
 		    ioc, MPI_SAS_OP_CLEAR_ALL_PERSISTENT);
 	}
 
-	add_taint(TAINT_SUPPORT_REMOVED, LOCKDEP_STILL_OK);
-	pr_warn("MPTSAS MODULE IS NOT SUPPORTED\n");
-
 	error = scsi_add_host(sh, &ioc->pcidev->dev);
 	if (error) {
 		dprintk(ioc, printk(MYIOC_s_ERR_FMT
@@ -5356,8 +5353,18 @@ static void mptsas_remove(struct pci_dev
 }
 
 static struct pci_device_id mptsas_pci_table[] = {
+	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_SAS1064,
+		PCI_ANY_ID, PCI_ANY_ID },
 	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_SAS1068,
-		PCI_VENDOR_ID_VMWARE, PCI_ANY_ID },
+		PCI_ANY_ID, PCI_ANY_ID },
+	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_SAS1064E,
+		PCI_ANY_ID, PCI_ANY_ID },
+	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_SAS1068E,
+		PCI_ANY_ID, PCI_ANY_ID },
+	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_SAS1078,
+		PCI_ANY_ID, PCI_ANY_ID },
+	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_SAS1068_820XELP,
+		PCI_ANY_ID, PCI_ANY_ID },
 	{0}	/* Terminating entry */
 };
 MODULE_DEVICE_TABLE(pci, mptsas_pci_table);
diff -Naurp rhel8-4.18.0-80/drivers/message/fusion/mptspi.c kernel-4.20.18/drivers/message/fusion/mptspi.c
--- a/drivers/message/fusion/mptspi.c	2019-03-13 11:04:53.000000000 +0000
+++ b/drivers/message/fusion/mptspi.c	2018-11-21 08:22:14.000000000 +0000
@@ -1245,7 +1245,11 @@ static struct spi_function_template mpts
 
 static struct pci_device_id mptspi_pci_table[] = {
 	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_53C1030,
-		PCI_VENDOR_ID_VMWARE, PCI_ANY_ID },
+		PCI_ANY_ID, PCI_ANY_ID },
+	{ PCI_VENDOR_ID_ATTO, MPI_MANUFACTPAGE_DEVID_53C1030,
+		PCI_ANY_ID, PCI_ANY_ID },
+	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_53C1035,
+		PCI_ANY_ID, PCI_ANY_ID },
 	{0}	/* Terminating entry */
 };
 MODULE_DEVICE_TABLE(pci, mptspi_pci_table);
@@ -1536,10 +1540,6 @@ mptspi_probe(struct pci_dev *pdev, const
 		    0, 0, 0, 0, 5);
 
 	scsi_scan_host(sh);
-
-	add_taint(TAINT_SUPPORT_REMOVED, LOCKDEP_STILL_OK);
-	pr_warn("MPTSPI MODULE IS NOT SUPPORTED\n");
-
 	return 0;
 
 out_mptspi_probe:
